### ISSUE TITLE
Multiple Environments & TF State

### ISSUE THREAD

**Issue by fulhaq** [rgl/terraform-proxmox-talos#6]
I want to be able to run this code across multiple proxmox clusters namely (prod, staging, qa). I can do it by tinkering with state files, etc. but was really hoping for a more robust way of doing this. It may be my lack of understanding of how TF handles envs. I come from terragrunt (TG) background and that's the only way I can think of handling this. Thoughts on TF envs vs TG?

Thank you!

**Comment by rgl** [comment]
AFAICT, terraform oss still (?) does not handle that. we have to manually configure and run it against different terraform state locations or use something like terragrunt or terramate.

please, keep in mind that this repo lacks production features (e.g. it does not handle multiple envs / ha / backups / upgrades / etc), this is meant to be a simple talos playground.

**Comment by fulhaq** [comment]
Yeah a terragrunt introduction would be great to this project! Understood on the playground but I'd like to use it for my homeserver "production" use. wouldn't mind collaborating on it with you if you're open to it.

Envs: Terragrunt
HA: If this code can create talos nodes on multiple proxmox nodes, we may inch closer to achieving HA.
Backups: I've played around with Veeam/Velero in the past. I'd be interested in adding it to the cluster.
Observability: It should be easy enough to add prom-kube-stack via helm.
Upgrades: I agree the way the code stands now, it' won't upgrade clusters gracefully. Worth looking into.

I can get started on some of these if you'd be interested. 

Thanks!
